### PR TITLE
Tracing Helm followup and documentation

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Update documentation around new tracing configuration

# Context

Follow up PR to: https://github.com/solo-io/gloo/pull/3838

An upgrade to the envoy API has changed the way users configure tracing in their clusters. Update the documentation and helm values to demonstrate this change.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3900

### CODE REVIEW

File: docs/content/guides/observability/tracing.md [hunk 1]
```diff
@@ -19,37 +19,25 @@ Gloo Edge makes it easy to implement tracing on your system through [Envoy's tra
...
+1. Configure a tracing provider
 1. (Optional) Annotate routes with descriptors
 
-##### 1. Specify a tracing provider in the bootstrap config
+##### 1. Configure a tracing cluster
 
-The bootstrap config is the portion of Envoy's config that is applied when an Envoy process in intialized.
+Tracing requires a cluster that will collect the traces. For example, Zipkin requires a `collector_cluster` to be specified in the bootstrap config. If your provider requires a cluster to be specified, you can provide it in the config, as shown below.
+
+The bootstrap config is the portion of Envoy's config that is applied when an Envoy process in initialized.
```
Comment (kdorosh) [inline docs/content/guides/observability/tracing.md:31]: typo, 
```suggestion
The bootstrap config is the portion of Envoy's config that is applied when an Envoy process is initialized.
```

File: docs/content/guides/observability/tracing.md [hunk 2]
```diff
@@ -139,6 +103,69 @@ data:
                       port_value: 1234
 {{< /highlight >}}
 
+To apply the bootstrap config to Envoy we need to restart the process. An easy way to do this is with `kubectl delete pod`.
+
+```bash
+kubectl delete pod -n gloo-system gateway-proxy-[suffix]
```
Comment (kdorosh) [inline docs/content/guides/observability/tracing.md:109]: k8s 1.15 added a nicer command for this, `kubectl rollout restart deployment [deployment_name]`

File: docs/content/guides/observability/tracing.md [hunk 3]
```diff
@@ -139,6 +103,69 @@ data:
...
+When the `gateway-proxy` pod restarts it should have the new trace cluster config.
+
+##### 2. Configure a tracing provider
+
+For a list of supported tracing providers, and the configuration that they expect, please see Envoy's documentation on [trace provider configuration](https://www.envoyproxy.io/docs/envoy/v1.13.1/api-v2/config/trace/v2/trace.proto#config-trace-v2-tracing-http).
+For demonstration purposes, we show how to configure a *zipkin* trace provider below.
+
+**Option 1 (Preferred): Set the tracing provider on a dynamic listener:**
+
+You can enable tracing on a listener-by-listener basis. Gloo Edge exposes this feature through a listener plugin. Please see [the tracing listener plugin docs]({{% versioned_link_path fromRoot="/guides/traffic_management/listener_configuration/http_connection_manager/#tracing" %}}) for details on how to enable tracing on a listener.
```
Comment (kdorosh) [inline docs/content/guides/observability/tracing.md:121]: let's avoid the word "plugin" in our docs. It doesn't mean much to users who aren't familiar with the codebase, and using it in the past has mislead users into thinking that this was support for a compiled external binary to work with Gloo Edge (similar to our go extauth plugins, which do exactly that: https://docs.solo.io/gloo-edge/latest/guides/security/auth/extauth/plugin_auth/)

File: docs/content/guides/observability/tracing.md [hunk 4]
```diff
@@ -139,6 +103,69 @@ data:
...
+                        name: envoy.tracers.zipkin
+                        typed_config:
+                          "@type": "type.googleapis.com/envoy.config.trace.v2.ZipkinConfig"
+                          collector_cluster: zipkin
+                          collector_endpoint: "/api/v2/spans"
+                          collector_endpoint_version: HTTP_JSON
+{{< /highlight >}}
+
 
 To apply the bootstrap config to Envoy we need to restart the process. An easy way to do this is with `kubectl delete pod`.
```
Comment (kdorosh) [inline docs/content/guides/observability/tracing.md:170]: again `kubectl rollout restart` might be a bit nicer

**Review by kdorosh — COMMENTED** [review]
looks great, just a couple docs nits

File: docs/content/guides/observability/tracing.md (hunk 3 above)
Comment (sam-heilbron) [reply docs/content/guides/observability/tracing.md:121]: Yeah that makes sense. I was trying to maintain the existing sentiment of the docs, but it makes sense that overusing plugin will just confuse users.

**Review by kdorosh — APPROVED** [review]

**Pull request merged**